test/helm_task_test.go

```diff
@@ -0,0 +1,310 @@
+	}
+
+	// Verify status of PipelineRun (wait for it)
+	if err := WaitForPipelineRunState(c, helmDeployPipelineRunName, func(pr *v1alpha1.PipelineRun) (bool, error) {
+		c := pr.Status.GetCondition(duckv1alpha1.ConditionSucceeded)
+		if c != nil {
+			if c.Status == corev1.ConditionTrue {
+				return true, nil
+			} else if c.Status == corev1.ConditionFalse {
+				return true, fmt.Errorf("pipeline run %s failed!", helmDeployPipelineRunName)
+			}
+		}
+		return false, nil
+	}, "PipelineRunCompleted"); err != nil {
+		t.Errorf("Error waiting for PipelineRun %s to finish: %s", helmDeployPipelineRunName, err)
+	}
+
+	k8sService, err := c.KubeClient.Kube.CoreV1().Services(namespace).Get(helmDeployServiceName, metav1.GetOptions{})
+	if err != nil {
+		t.Errorf("Error getting service at %s %s", helmDeployServiceName, err)
+	}
+	var serviceIp string
+	ingress := k8sService.Status.LoadBalancer.Ingress
+	if len(ingress) > 0 {
+		serviceIp = ingress[0].IP
+	}
+
+	resp, err := http.Get(fmt.Sprintf("http://%s:8080", serviceIp))
+	if err != nil {
+		t.Errorf("Error reaching service at http://%s:8080 %s", serviceIp, err)
+	}
+	if resp != nil && resp.StatusCode != http.StatusOK {
+		t.Errorf("Error from service at http://%s:8080 %s", serviceIp, err)
+	}
+}
+
+func getGoHelloworldGitResource(namespace string) *v1alpha1.PipelineResource {
+	return &v1alpha1.PipelineResource{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      sourceResourceName,
+			Namespace: namespace,
+		},
+		Spec: v1alpha1.PipelineResourceSpec{
+			Type: v1alpha1.PipelineResourceTypeGit,
+			Params: []v1alpha1.Param{
+				v1alpha1.Param{
+					Name:  "Url",
+					Value: "https://github.com/pivotal-nader-ziada/gohelloworld",
+				},
+			},
+		},
+	}
+}
+
+func getCreateImageTask(namespace string, t *testing.T) *v1alpha1.Task {
+	// according to knative/test-infra readme (https://github.com/knative/test-infra/blob/13055d769cc5e1756e605fcb3bcc1c25376699f1/scripts/README.md)
+	// the KO_DOCKER_REPO will be set with according to the porject where the cluster is created
+	// it is used here to dunamically get the docker registery to push the image to
+	dockerRepo := os.Getenv("KO_DOCKER_REPO")
+	if dockerRepo == "" {
+		t.Fatalf("KO_DOCKER_REPO env variable is required")
+	}
+
+	imageName = fmt.Sprintf("%s/%s", dockerRepo, sourceImageName)
+
+	return &v1alpha1.Task{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: namespace,
+			Name:      createImageTaskName,
+		},
+		Spec: v1alpha1.TaskSpec{
+			Inputs: &v1alpha1.Inputs{
+				Resources: []v1alpha1.TaskResource{
+					v1alpha1.TaskResource{
+						Name: sourceResourceName,
+						Type: v1alpha1.PipelineResourceTypeGit,
+					},
+				},
+			},
+			BuildSpec: &buildv1alpha1.BuildSpec{
+				Steps: []corev1.Container{{
+					Name:  "kaniko",
+					Image: "gcr.io/kaniko-project/executor",
+					Args: []string{"--dockerfile=/workspace/Dockerfile",
+						fmt.Sprintf("--destination=%s", imageName),
+					},
```

Comment:
in the long run i feel like we can probably get rid of the kaniko specific integration test since we're using kaniko here too. @tanner-bruce is making some updates to it in #158 that we don't want to lose tho, so maybe we can do that in another PR later sometime.

Reply:
yes, agreed